types/cassandra-driver/index.d.ts

```diff
@@ -889,3 +896,246 @@ export namespace metadata {
+  getLoadBalancingPolicy(): policies.loadBalancing.LoadBalancingPolicy;
+  getPageState(): Buffer;
+  getRawQueryOptions(): QueryOptions;
+  getReadTimeout(): number;
+  getRetryPolicy(): policies.retry.RetryPolicy;
+  getRoutingKey(): Buffer | Buffer[];
+  getSerialConsistency(): types.consistencies;
+  getTimestamp(): number | Long | undefined | null;
+  setHints(hints: string[]): void;
+}
+
+export let ExecutionOptions: ExecutionOptionsStatic;
+
+export namespace mapping {
+  let Mapper: MapperStatic;
+  let ModelBatchItem: ModelBatchItemStatic;
+  let ModelMappingInfo: ModelMappingInfoStatic;
+  let ModelMapper: ModelMapperStatic;
+  let MappingHandler: MappingHandlerStatic;
+  let Result: ResultStatic;
+  let DefaultTableMappings: DefaultTableMappingsStatic;
+  let UnderscoreCqlToCamelCaseMappings: UnderscoreCqlToCamelCaseMappingsStatic;
+
+  interface MappingExecutionOptions {
+    executionProfile?: string;
+    isIdempotent?: boolean;
+    logged?: boolean;
+    timestamp?: number | Long;
+    fetchSize?: number;
+    pageState?: number;
+  }
+
+  interface ModelTables {
+    name: string;
+    isView: boolean;
+  }
+
+  interface MappingQuery {
+    query: string;
+    isIdempotent: boolean;
+    isCounter: boolean;
+  }
+
+  interface MapperStatic {
+    new (client: Client, options?: MappingOptions): Mapper;
+  }
+  interface Mapper {
+    batch(items: ModelBatchItem[], executionOptions?: string | MappingExecutionOptions): Promise<Result>;
+    forModel(name: string): ModelMapper;
+  }
+
+  interface MappingOptions {
+    models: { [key: string]: ModelOptions };
+  }
+
+  interface ModelOptions {
+    tables?: string[] | ModelTables[];
+    mappings?: TableMappings;
+    columns?: { [key: string]: string };
+    keyspace?: string;
+  }
+
+  interface ModelMappingInfoStatic {
+    new (keyspace: string, tables: ModelTables[], mappings: TableMappings, columns: { [key: string]: string }): ModelMappingInfo;
+    parse(options: MappingOptions, currentKeyspace: string): { [key: string]: ModelMappingInfo };
+    createDefault(modelName: string, currentKeyspace: string): ModelMappingInfo;
+  }
+  interface ModelMappingInfo {
+    keyspace: string;
+    tables: ModelTables[];
+
+    getColumnName(propName: string): string;
+    getPropertyName(columnName: string): string;
+    newInstance(): TableMappings;
+  }
+
+  interface MappingDocInfo {
+    fields?: string[];
+    ttl?: number;
+    ifNotExists?: boolean;
+    when?: { [key: string]: string };
+    orderBy?: { [key: string]: string };
+    limit?: number;
+    deleteOnlyColumns?: boolean;
+  }
+
+  interface ModelBatchItemStatic {
+    new (
+      queries: Promise<MappingQuery[]>,
+      doc: { [key: string]: any },
+      docInfo: MappingDocInfo,
+      mappingInfo: ModelMappingInfo
+    ): ModelBatchItem;
+  }
+  interface ModelBatchItem {
+    pushQueries(arr: MappingQuery[]): { isIdempotent: boolean, isCounter: boolean };
+    getMappingInfo(): ModelMappingInfo;
```

Comment:
These two functions are marked as *internal*.
- https://github.com/datastax/nodejs-driver/compare/v3.6.0...v4.0.0#diff-d64b42be2e4cb54953df1ab1fdb3ed68R23
- https://github.com/datastax/nodejs-driver/compare/v3.6.0...v4.0.0#diff-d64b42be2e4cb54953df1ab1fdb3ed68R48
I'm not sure if we should then add them to the type definition.

Reply:
You are absolutely right, I don't know how I missed those tags. They are not in the docs either.
Removed them.